addons/mail/static/src/model/model_manager.js

```diff
@@ -1075,6 +1081,14 @@ class ModelManager {
             if (!field) {
                 throw new Error(`Cannot create/update record with data unrelated to a field. (model: "${Model.modelName}", non-field attempted update: "${fieldName}")`);
             }
+
+            if ((!options.allowWriteReadonly) && field.readonly) {
+                throw new Error(`Can't update "${field.fieldName}" (model: "${Model.modelName}") because it's readonly.`);
```

Comment:
instead of telling model, error message would be more useful if it gave record localId (feel free to also update the error above)